src/util/arith_tools.cpp

```diff
@@ -111,6 +111,29 @@ bool to_integer(const constant_exprt &expr, mp_integer &int_value)
 
 /*******************************************************************\
 
+Function: to_unsigned_integer
+
+  Inputs: a constant expression and a reference to an unsigned int
+
+ Outputs: an error flag
+
+ Purpose: convert a positive integer expression to an unsigned int
+
+\*******************************************************************/
+
+bool to_unsigned_integer(const constant_exprt &expr, unsigned &uint_value)
+{
+  mp_integer i;
+  if(to_integer(expr, i))
+    return true;
+  if(i<0)
+    return true;
+  else
+    return integer2unsigned(i);
```

Comment:
This should be:
```
else
{
  uint_value=integer2unsigned(i);
  return false;
}
```
(and now I'm wondering whether there was any test covering this?)